public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,903 @@
+  map to the properties of the _form model_.
+
+  2. User changes flow from the DOM elements to the _form model_, not to the _data model_.
+  The form controls never update the _data model_.
+
+  The _form_ and _data_ model structures need not match exactly.
+  You often present a subset of the _data model_ on a particular screen.
+  But it makes things easier if the shape of the _form model_ is close to the shape of the _data model_.
+  
+  In this `HeroDetailComponent`, the two models are quite close.
+
+  Recall the definition of `Hero` in `data-model.ts`:
++makeExample('reactive-forms/ts/app/data-model.ts', 'model-classes','app/data-model.ts (classes)')(format=".")
+:marked
+  Here, again, is the component's `FormGroup` definition.
+
++makeExample('reactive-forms/ts/app/hero-detail-5.component.ts', 'hero-form-model','app/hero-detail.component.ts (excerpt)')(format=".")
+:marked
+  There are two significant differences between these models:
+
+  1. The `Hero` has an `id`. The form model does not because you generally don't show primary keys to users.
+
+  1. The `Hero` has an array of addresses. This form model presents only one address,
+  a choice [revisited below](#form-array "Form arrays").
+
+  Nonetheless, the two models are pretty close in shape and you'll see in a moment how this alignment facilitates copying the _data model_ properties
+  to the _form model_ with the `patchValue` and `setValue` methods.
+
+:marked
+  Take a moment to refactor the _address_ `FormGroup` definition for brevity and clarity as follows:
++makeExample('reactive-forms/ts/app/hero-detail-6.component.ts', 'address-form-group')(format=".")
+
+:marked
+  Also be sure to update the import from `data-model` so you can reference the `Hero` class:
++makeExample('reactive-forms/ts/app/hero-detail-5.component.ts', 'import-hero')(format=".")
+
+.l-main-section
+a#set-data
+:marked
+  ## Populate the form model with _setValue_ and _patchValue_
+  Previously you created a control and initialized its value at the same time. 
+  You can also initialize or reset the values _later_ with the 
+  `setValue` and `patchValue` methods.
+
+  ### _setValue_
+  With **`setValue`**, you assign _every_ form control value _at once_
+  by passing in a data object whose properties exactly match the _form model_ behind the `FormGroup`.
+
++makeExample('reactive-forms/ts/app/hero-detail-6.component.ts', 'set-value','app/hero-detail.component.ts (excerpt)')(format=".")
+:marked
+  The `setValue` method checks the data object thoroughly before assigning any form control values.
+
+  - It rejects a data object that doesn't match the `FormGroup` structure.
+  - It rejects a data object if it is missing values for any control in the group.
+  - It returns helpful error messages.
+
+  If you have a typo or nested controls incorrectly, `setValue` will catch 
+  the error and report it clearly. `patchValue` will fail silently.
```

Comment:
Reading through this again, I think we should replace the three points above with a few sentences. The bullet points are a bit odd.

"It will not accept a data object that doesn't match the `FormGroup` structure or is missing values for any control in the group. This way, it can return helpful error messages if you have a typo or if you've nested controls incorrectly. `patchValue` will fail silently."